getting-started/git-boot-camp.rst

```diff
@@ -286,11 +286,24 @@ Creating a Pull Request
 You should include the issue number in the title of the PR,
 in the format ``gh-NNNNN: <PR Title>``.
 
-You might also use `special keywords
-<https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword>`_
-to automatically close the linked issue once the PR is merged. However, you
-should avoid them if other PRs need to be merged before the issue can be
-closed (e.g. backports to other branches).
+Linking to Issues and Pull Requests
+-----------------------------------
+
+You can link to issues and pull requests using ``gh-NNNNN`` (this form is
+preferred over ``#NNNNN``).  If the reference appears in a list, the link
+will be expanded to show the status and title of the issue/PR.
```

Comment:
Is that really what triggers this? I've gotten inconsistent results in my resent testing, with links sometimes expanding and sometimes not in and outside of lists, even in the same spot. I'd be very curious to know.

Also, `gh` (and `#`?) references in commit messages get expanded to `pythongh`—how does that happen?

Reply:
> Oh, is that really what triggers this?

I think so, and it took me a while to figure that out too.  ISTM that references like `#855` are consistently expanded in lists like:
* #855 
* with some text #855
1. #855
2. with some text #855
- [ ] #855
- [ ] with some text #855

But not in paragraphs like this (#855) or quotes like

> #855

Do you have any counterexamples?

> Also, gh (and #?) references in commit messages get expanded to pythongh—how does that happen?

I've seen that and I think it's just a bug in GitHub.

Reply:
> with some text https://github.com/python/devguide/pull/855

Ah, I misinterpreted what you said originally to mean just bullets with no other text, when I was seeing them both with and without—silly me.

> Do you have any counterexamples?

Not any others that I can reliably reproduce—I believe the behavior I saw (on the What's New issues) when I was trying to figure this out might have perhaps just been a combination of an issue where the link wasn't getting created as above when the linked issue hadn't been created when I loaded the page, and/or just not being totally sure what caused it when testing a number of things (particularly `#NNN` vs `org/repo#NNN` vs `<full URL>` vs. <comment URL>).

Reply:
One side note that came up during some related testing—AFAIK, `GH-`/`gh-` doesn't trigger the issue number lookup and autocompletion, which I find invaluable especially on repos nearing 6 digit issue and PR numbers like CPython.

Reply:
In https://github.com/python/devguide/pull/946#pullrequestreview-1096790696 the ref is not expanded, maybe because it's in a review?

Screenshot showing both the unexpanded list, and the popup on the first ref which confirms the ref is correct:
![image](https://user-images.githubusercontent.com/25624924/194470794-9b80c29a-85f2-42d4-8ca9-62d2466957dc.png)